dashboard/client/src/pages/job/TaskTimeline.tsx

```diff
@@ -0,0 +1,231 @@
+        setLoading(false);
+        return;
+      }
+
+      if (evt.data !== "PONG") {
+        return;
+      }
+
+      window.clearInterval(timer);
+      win.postMessage(arrayBuffer, PERFETTO_URL);
+      // We now finished uploading timeline.
+      setLoading(false);
+    };
+
+    const onDialogClose = () => {
+      setOpenDialog(false);
+    };
+
+    const onClick = () => {
+      setOpenDialog(true);
+    };
+
+    const onClickPerfettoLoading = () => {
+      if (!loading) {
+        setLoading(true);
+        // Instead of downloading a json file, it posts the result to
+        // perfetto directly.
+        getTaskTimeline(jobId)
+          .then((resp) => {
+            openTrace(resp.data);
+          })
+          .catch((error) => {
+            setLoading(false);
+            console.log(
+              `Failed to fetch data for the timeline. Error: ${error}`,
+            );
+          });
+      }
+      setOpenDialog(false);
+    };
+
+    // TODO(sang): Raise a error message when fetch fails.
+    if (openDialog) {
+      return (
+        <AlertDialog
+          open={openDialog}
+          onAgree={onClickPerfettoLoading}
+          handleClose={onDialogClose}
+          title={"View on ui.perfetto.dev (a Google service)"}
+          contents={
+            "Timeline tracing will be uploaded to ui.perfetto.dev for tracing visualization. According to https://perfetto.dev/docs/#trace-visualization, the data is not suppposed to be uploaded to their server."
+          }
+        />
+      );
+    } else if (loading) {
+      return (
+        <div>
+          <Button variant="outlined" disabled={loading}>
+            Open Timeline View <CircularProgress color="primary" size={16} />
+          </Button>
+        </div>
+      );
+    } else {
+      return (
+        <Button variant="outlined" disabled={loading} onClick={onClick}>
+          Open Timeline View
+        </Button>
+      );
+    }
+  };
+
+  return (
+    <div className={classes.root}>
+      <Grid container alignItems="center" spacing={4}>
+        <Grid item>
+          Timeline view shows how many tasks are scheduled and executed across
+          the cluster over time.
+        </Grid>
+      </Grid>
+      <Grid container alignItems="center" spacing={4}>
+        <Grid item>
+          <TimelineDownloadButton jobId={jobId} />
+        </Grid>
+        <Grid item>
+          <TimelineButton jobId={jobId} />
+        </Grid>
+      </Grid>
+      <Grid container alignItems="center" spacing={2}>
+        <Grid item>
+          <Accordion>
+            <AccordionSummary
+              expandIcon={<ExpandMoreIcon />}
+              aria-controls="panel1a-content"
+              id="panel1a-header"
+            >
+              <Typography variant="h6">Instruction</Typography>
+            </AccordionSummary>
+            <AccordionDetails>
+              <Typography variant="subtitle1">
+                <List>
```

Comment:
Need help in design improvement here

Reply:
Do we have a UX mockup for this?

Reply:
We don't. That's why it looks horrible. cc @scottsun94 for mockup

Reply:
I think we have to have this instruction as download button is there